### PR TITLE
Implement matrix from (anti)diagonals

### LINKED ISSUE #1240: Add elements to get matrix from list of diagonals/anti-diagonals

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #1240

### CODE REVIEW

**Comment by Lyxal** [comment]
There's a chance I've gotten this wrong. Sorry about that if that's the case.

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -6345,6 +6345,27 @@
...
 
+- element: "Þ‟"
+  name: List from Anti-Diagonals
+  description: List of lists from anti-diagonals of a list
+  arity: 1
+  overloads:
+    lst: List of lists from anti-diagonals of a list
+  vectorise: false
+  tests:
+    - "[[[1], [2, 6], [3, 7, 11], [4, 8, 12, 16], [5, 9, 13, 17, 21], [10, 14, 18, 22], [15, 19, 23], [20, 24], [25]]] : [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14, 15], [16, 17, 18, 19, 20], [21, 22, 23, 24, 25]]"
```
Comment (ysthakur) [inline documents/knowledge/elements.yaml:6356]: Maybe a test case for a non-square matrix?

**Review by ysthakur — CHANGES_REQUESTED** [review]

File: vyxal/elements.py [hunk 2]
```diff
@@ -3562,6 +3562,38 @@ def gen():
...
+@element("Þ‟", 1)
+def list_from_anti_diagonals(lhs, ctx):
+    """ELement Þ‟
+    (lst) -> Turn a list of anti-diagonals into a list of lists. Expects shortest anti-diagonal to be first.
+    """
+
+    f = lambda x: map(
+        lambda a: [x[z[-1] + b - a][min(a, b)] for b in z],
+        z := range(len(x) // 2 + 1),
+    )
```
[the remark this answers is not in the document]
Comment (Lyxal) [reply vyxal/elements.py:3574]: I don't understand it at all myself either. I found it on a code golf challenge :p

**Review by ysthakur — CHANGES_REQUESTED** [review]

File: vyxal/elements.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Steffan153) [reply vyxal/elements.py]: Really? `map` is really ungolfy with lambdas. Maybe it was just a bad golfer

File: documents/knowledge/elements.yaml (hunk 1 above)
Comment (Lyxal) [reply documents/knowledge/elements.yaml:6356]: Pretty sure this only really applies to square matrixes

**Pull request merged**